### PR TITLE
FAT-36 UI tests replacement with RTL/Jest

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/FAT-36
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by NikitaSedyx — APPROVED** [review]

File: src/OrdersList/OrderExportSettingsModalContainer.test.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 jest.mock('../common/ExportSettingsModal/ExportSettingsModalContainer', () => jest.fn().mockReturnValue('ExportSettingsModalContainer'));
 
-jest.mock('../common/utils', () => ({
-  ...jest.requireActual('../common/utils'),
-  fetchExportDataByIds: jest.fn(),
-}));
+// jest.mock('../common/utils', () => ({
+//   ...jest.requireActual('../common/utils'),
+//   // fetchExportDataByIds: jest.fn(),
+// }));
```
Comment (AndreiShumski2) [inline src/OrdersList/OrderExportSettingsModalContainer.test.js:18]: ?

**Review by AndreiShumski2 — APPROVED** [review]

File: src/OrdersList/OrderExportSettingsModalContainer.test.js (hunk 1 above)
Comment (usavkov-epam) [reply src/OrdersList/OrderExportSettingsModalContainer.test.js]: removed

File: src/OrderLinesList/Details/OrderLineDetails.test.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  });
+});
+
+describe('OrderLineDetails actions', () => {
+  beforeEach(() => {
+    POLineView.mockClear();
+  });
+
+  it('should navigate to order details view', async () => {
+    await waitFor(() => renderOrderLineDetails());
```
Comment (axelhunn) [inline src/OrderLinesList/Details/OrderLineDetails.test.js:76]: IMHO it's not idiomatic to await for render

File: src/OrderLinesList/LineExportSettingModalContainer.test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  it('should render Export Settings Modal Container', () => {
+    renderLineExportSettingModalContainer();
+
+    expect(screen.getByText('ExportSettingsModalContainer')).toBeInTheDocument();
+  });
+
+  it('should call fetchAllRecords function if exporting', async () => {
+    renderLineExportSettingModalContainer();
+
+    await act(() => ExportSettingsModalContainer.mock.calls[0][0].fetchOrdersAndLines());
```
Comment (axelhunn) [inline src/OrderLinesList/LineExportSettingModalContainer.test.js:50]: `await act` looks not so obvious to me too.
If we keep to "arrange-act-assert" pattern, it should be here more like:
- act
- then await for something to appear (or be called)

File: src/OrderLinesList/Details/OrderLineDetails.test.js (hunk 2 above)
Comment (usavkov-epam) [reply src/OrderLinesList/Details/OrderLineDetails.test.js]: changed

File: src/OrderLinesList/LineExportSettingModalContainer.test.js (hunk 3 above)
Comment (usavkov-epam) [reply src/OrderLinesList/LineExportSettingModalContainer.test.js]: yeah, thanks, I changed it

**Review by axelhunn — APPROVED** [review]

**Pull request merged**